jwst/tweakreg/astrometric_utils.py

```diff
@@ -0,0 +1,179 @@
+        catalog web service.  If None, no file will be written out.
+
+    gaia_only : bool, optional
+        Specify whether or not to only use sources from GAIA in output catalog
+
+    existing_wcs : `~stwcs.wcsutil.HSTWCS`
+        existing WCS object specified by the user
+
+    num_sources : int
+        Maximum number of brightest/faintest sources to return in catalog.
+        If `num_sources` is negative, return that number of the faintest
+        sources.  By default, all sources are returned.
+
+    Notes
+    -----
+    This function will point to astrometric catalog web service defined
+    through the use of the ASTROMETRIC_CATALOG_URL environment variable.
+
+    Returns
+    -------
+    ref_table : `~astropy.table.Table`
+        Astropy Table object of the catalog
+
+    """
+
+    # start by creating a composite field-of-view for all inputs
+    # This default output WCS will have the same plate-scale and orientation
+    # as the first chip in the list, which for WFPC2 data means the PC.
+    # Fortunately, for alignment, this doesn't matter since no resampling of
+    # data will be performed
+    if existing_wcs is not None:
+        outwcs = existing_wcs
+    else:
+        outwcs = resample_utils.make_output_wcs(input_models)
+    radius = compute_radius(outwcs)
+    ra, dec = outwcs.wcs.crval
+
+    # perform query for this field-of-view
+    ref_dict = get_catalog(ra, dec, sr=radius, catalog=catalog)
+    colnames = ('ra', 'dec', 'mag', 'objID')
+
+    ref_table = ref_dict[colnames]
+
+    # Add catalog name as meta data
+    ref_table.meta['catalog'] = catalog
+    ref_table.meta['gaia_only'] = gaia_only
+
+    # rename coordinate columns to be consistent with tweakwcs
+    ref_table.rename_column('ra', 'RA')
+    ref_table.rename_column('dec', 'DEC')
+
+    # Append GAIA ID as a new column to the table...
+    gaia_sources = []
+    for source in ref_dict:
+        if 'GAIAsourceID' in source:
+            g = source['GAIAsourceID']
+            if gaia_only and g.strip() == '':
+                continue
+        else:
+            g = "-1"  # indicator for no source ID extracted
+        gaia_sources.append(g)
+
+    gaia_col = table.Column(data=gaia_sources, name='GaiaID', dtype='U25')
+    ref_table.add_column(gaia_col)
+
+    # sort table by magnitude, fainter to brightest
+    ref_table.sort('mag', reverse=True)
+
+    if num_sources is not None:
+        indx = -1 * num_sources
+        ref_table = ref_table[:indx] if num_sources < 0 else ref_table[indx:]
+
+    # Write out table to a file, if specified
+    if output is not None:
+        ref_table.write(output, format=table_format, overwrite=True)
+
+    return ref_table
+
+
+"""
+
+Utility functions for creating an astrometric reference catalog.
+
+"""
+
+def compute_radius(wcs):
+    """Compute the radius from the center to the furthest edge of the WCS."""
+
+    ra, dec = wcs.wcs.crval
```

Comment:
This assumes FITS WCS which is probably not correct.

Reply:
Again, this has been revised using the same information that assign_wcs relies on for its computations.